### ISSUE TITLE
Failed to collect tests on Windows (tests are collected OK on Linux or macOS)

### ISSUE THREAD

**Issue by pombredanne** [pytest-dev/pytest#4006]
In the [scancode-toolkit](https://github.com/nexB/scancode-toolkit), I run a fairly large tests suite on Windows, Linux and macOS (with Travis and Appveyor).
I just noticed that some tests are NOT collected at all on Windows on Appveyor (and I confirmed this on plain Windows VM).
Here are two output of running the same --collect-only command on both OSes:
- `py.test -vvs tests/licensedcode/test_detection_datadriven.py --collect-only` [linux-ldd.txt](https://github.com/pytest-dev/pytest/files/2400512/linux-ldd.txt) has 2000+ tests collected
- `py.test -vvs  --collect-only   tests\licensedcode\test_detection_datadriven.py > windows-ldd.txt` [windows-ldd.txt](https://github.com/pytest-dev/pytest/files/2400513/windows-ldd.txt) has none of these

The thing about theses tests is that these are parametric tests (not using py.test features for this) where tests methods [are attached](https://github.com/nexB/scancode-toolkit/blob/67d727aa54e0e7fe3588a89a123a20f3c3f76da1/tests/licensedcode/test_detection_datadriven.py#L178) to a test class at [import time](https://github.com/nexB/scancode-toolkit/blob/67d727aa54e0e7fe3588a89a123a20f3c3f76da1/tests/licensedcode/test_detection_datadriven.py#L300) .

There are other places where similar tests are [collected alright](https://github.com/nexB/scancode-toolkit/blob/67d727aa54e0e7fe3588a89a123a20f3c3f76da1/tests/cluecode/test_copyrights.py#L54).

To reproduce this issue :
- ensure that you have a Python 2.7 installation
- clone or download [scancode-toolkit](https://github.com/nexB/scancode-toolkit)
- run `./configure` (or `configure`  on Windows): this will create a local virtualenv and install all the bundled deps at once
- on POSIX, run `source bin/activate`
- from that point on you are in a virtualenv with py.test installed and you can issue the py.test commands above.

The default py.test version used ATM is [3.5.1](https://github.com/nexB/scancode-toolkit/blob/develop/thirdparty/dev/pytest.ABOUT)  

I tested the bug also with the latest 3.8.0 on windows and that's the same problem.

**Comment by pombredanne** [comment]
@nicoddemus any idea of what would be the root cause for this?

**Comment by nicoddemus** [comment]
Hi @pombredanne,

Sorry for the delay, this has slipped through the cracks it seems.

I've debugged the code and noticed this is not a pytest problem, because I can reproduce the problem without pytest:

```python
from unittest import TestLoader
import test_detection_datadriven

loader = TestLoader()
for x in loader.getTestCaseNames(test_detection_datadriven.TestLicenseDataDriven):
    print(x)
```

This prints only:

```
test_detection
```

Digging further, I noticed that the tests methods being created by `build_tests` all had the same name (`test_detection`). This was caused by a naive `str.partition` call which used a hard coded unix path (`'licensedcode/data/'`).

I've opened https://github.com/nexB/scancode-toolkit/pull/1241 with a fix. 👍 

Closing this now as it is not a pytest issue.
